core/controllers/classifier.py

```diff
@@ -26,6 +31,17 @@
 from core.domain import email_manager
 import feconf
 
+_PARENT_DIR = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+_FUTURE_PATH = os.path.join(_PARENT_DIR, 'oppia_tools', 'future-0.17.1')
+
+sys.path.insert(0, _FUTURE_PATH)
+
+# pylint: disable=wrong-import-position
+# pylint: disable=wrong-import-order
+import past.builtins  # isort:skip
```

Comment:
`past.builtins` contains functions and attributes which are present in Python 2 but not in Python 3. For example, there is nothing called `basestring` in Py 3 so instead we need to write `past.builtins.basestring`.

`builtins` contains functions which are present in both Python versions but mean different things across the 2 versions. For example, `str(12)` in py 2 would return a string of value '12' but in py 3 it would return an unicode of value '12'. So we are specifying builtins.str everywhere in the codebase as we don't want to change their functioning.

Reply:
I think all this stuff should go in python_utils which has clear alternatives for py2 and py3. Swapping to py3 should then subsequently just entail changing a boolean flag.

Reply:
done

Reply:
EDIT: Its done only for the past.xx stuff. I am gonna do it for all the builtins in a follow up commit. Thanks!

Reply:
Done for all the builtins